provider/core/src/lib.rs

```diff
@@ -149,6 +149,7 @@ pub mod prelude {
     pub use crate::resource::ResourceKey;
     pub use crate::resource::ResourceOptions;
     pub use crate::resource::ResourcePath;
+    pub use crate::{leading_tag, resource_key, tagged, trailing_tag};
```

Comment:
Nit 1: Put these in the crate namespace but not the prelude.  Most people don't need these symbols, and when they do, it is often more readable to write them as `icu_provider::tagged!()` or similar.

Nit 2: There is no reason to export `leading_tag` and we should keep it private if possible.  Note in #1460 that I added a module `extract` in the core crate that reads these symbols so we don't export them.  We should export the function that does the work, in case we want to change the implementation again in the future.

Reply:
1. ok
2. But the 'function that does the work' is the `tagged` macro, so we need to export the tags.

Reply:
Nevermind the $crate prefix handles 2.

Reply:
Actually I think what's happening is you have `#[macro_export]` on the macros, which makes them importable from outside the crate.  It's annoying but I guess unavoidable.  I would have a slight preference for naming the tagging macros something like `__internal_leading_tag!()` or similar.  But this is something we should catch when we do the final documentation and API check before we actually ship 1.0.